File: 프로그래머스/lv2/64065. 튜플/튜플.java
import java.util.*;

class Solution {
    public List<Integer> solution(String s) {
        Map<Integer, Integer> hm = new HashMap<>();
        
        List<Integer> answer = new ArrayList<>();
        String[] str = s.split(",");        // 쉼표를 기준으로 분리
        
        // 중괄호 없애서 숫자만 남김
        for(int i=0; i<str.length; i++){
            str[i] = str[i].replace("{", "");
            str[i] = str[i].replace("}", "");
        }
        
        // 해시맵에 [숫자, 숫자 등장 횟수] 저장
        for(int i=0; i<str.length; i++){
            hm.put(Integer.parseInt(str[i]), hm.getOrDefault(Integer.parseInt(str[i]), 0)+1);
        }
        
        // 해시맵을 숫자 등장 횟수(value)를 기준으로 내림차순 정렬
        List<Integer> keySet = new ArrayList<>(hm.keySet());
        keySet.sort((o1, o2) -> hm.get(o2).compareTo(hm.get(o1)));    // 카운트 기준 내림차순 정렬
                   
        for(int key: keySet){
            answer.add(key);
        }
        
        return answer;
    }
}

File: 프로그래머스/2/64065. 튜플/튜플.py
def solution(s):
    answer = []
    
    # 집합은 원소의 순서가 바뀌므로
    # 튜플 (2, 1, 3, 4) -> {{1, 2, 3}, {2, 1}, {1, 2, 4, 3}, {2}}로 표현할 수도 있음
    # 1. 딕셔너리를 이용해 {원소:빈도수}를 저장하는 방법 -> 시간초과
    # 2. 길이가 작은 순서대로 발견한 원소를 집어 넣으면 튜플이 완성
        
    s = s.split('},{')
    for i in range(len(s)):
        s[i] = s[i].replace('{', '')
        s[i] = s[i].replace('}', '')

    s = sorted(s, key=lambda x:len(x))         # 길이가 작은 순 정렬

    count_set = set()
    
    for item in s:
        numbers = item.split(',')

        for num in numbers:
            if int(num) not in answer:
                answer.append(int(num))
        
    
    return answer